prow/spyglass/gcsartifact_fetcher_test.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package spyglass
+
+import (
+	"context"
+	"testing"
+)
+
+// Tests listing objects associated with the current job in GCS
+// TODO: fake-gcs-server does not support the GCS XML API, Issue #8923
+func testGCSListArtifacts(t *testing.T) {
+	blgArtifact := NewGCSArtifact(context.Background(), fakeGCSBucket.Object("logs/example-ci-run/403/build-log.txt"), "", "build-log.txt", 500e6)
+	srtArtifact := NewGCSArtifact(context.Background(), fakeGCSBucket.Object("logs/example-ci-run/403/started.json"), "", "started.json", 500e6)
+	finArtifact := NewGCSArtifact(context.Background(), fakeGCSBucket.Object("logs/example-ci-run/403/finished.json"), "", "finished.json", 500e6)
+	junitArtifact := NewGCSArtifact(context.Background(), fakeGCSBucket.Object("logs/example-ci-run/403/junit_01.xml"), "", "junit_01.xml", 500e6)
+	longLogArtifact := NewGCSArtifact(context.Background(), fakeGCSBucket.Object("logs/example-ci-run/403/long-log.txt"), "", "long-log.txt", 500e6)
+	testCases := []struct {
+		name              string
+		gcsJobSource      jobSource
+		expectedArtifacts []string
+	}{
+		{
+			name:         "Fetch Example CI Run #403 Artifacts",
```

Comment:
Test names should be useful for people investigating the test failure -- if this test fails, what does that mean? What functionality is testing? If this is a simple happy-case, write that.

Reply:
improved test case naming in latest